app/controllers/application_controller.rb

```diff
@@ -43,4 +45,15 @@ def destroy_user_if_current_user
     @user = User.new
   end 
   
+  private
+  
+  def extract_locale_from_accept_language_header
+    language = request.env['HTTP_ACCEPT_LANGUAGE'].presence
+    if language
+      locale = HttpAcceptLangParser.parse(language)
+      logger.debug "* Locale set using accept_language to '#{locale}'"
+      locale.in?(User::LANGUAGES) ? locale : nil
```

Comment:
Additionally: beware of comparing strings with symbols. What if tomorrow your code looked like this?

`LANGUAGES = %i[ar en].freeze`

(array of symbols)

Such a change is plausible, given that `I18n.locale` always returns a symbol, and as you gain experience you'll use symbols more frequently.

So, a safe comparison would be:

`locale.to_s.in?(User::LANGUAGES.map(&:to_s))`